Java API doesn't work with jdbc module

Both this code

```java
        var xtdb = IXtdb.startNode(NodeConfiguration.builder()
                .with("xtdb.jdbc/connection-pool", ModuleConfiguration.builder()
                        .with("dialect", ModuleConfiguration.builder()
                                .with("xtdb/module", "xtdb.jdbc.psql/->dialect")
                                .build()
                        )
                        .with(
                                "db-spec", ModuleConfiguration.builder()
                                        .with("jdbcUrl", JDBC_DATABASE_URL)
                                        .build()
                        ).build()
                )
                .with("xtdb/tx-log", ModuleConfiguration.builder()
                        .with("xtdb/module", "xtdb.jdbc/->tx-log")
                        .with("connection-pool", "xtdb.jdbc/connection-pool")
                        .build()
                )
                .with("xtdb/document-store", ModuleConfiguration.builder()
                        .with("xtdb/module", "xtdb.jdbc/->document-store")
                        .with("connection-pool", "xtdb.jdbc/connection-pool")
                        .build()
                ).build()
        );
```
        
And the equivalent using `Map.of` crash with 
        
```
Caused by: xtdb.IllegalArgumentException: Arg :db-spec = {jdbcUrl=jdbc:postgresql://127.0.0.1:5432/invite?user=postgres&password=postgres} invalid: {"jdbcUrl" "jdbc:postgresql://127.0.0.1:5432/invite?user=postgres&password=postgres"} - failed: map?
```
        
Which means that the module isn't happy with a Java style map, but there really isn't a good way to make a clojure map without janky interop

--- a/core/src/xtdb/api/ModuleConfiguration.java
+++ b/core/src/xtdb/api/ModuleConfiguration.java
@@ -10,7 +10,7 @@ import java.util.stream.Collectors;
 /**
  * Class to configure an XTDB module.
  *
- * See https://xtdb.com/reference/configuration.html for examples.
+ * See https://docs.xtdb.com/administration/configuring/ for examples.
  */
 @SuppressWarnings("unused") // entry points
 public final class ModuleConfiguration {

--- a/core/src/xtdb/api/NodeConfiguration.java
+++ b/core/src/xtdb/api/NodeConfiguration.java
@@ -9,7 +9,7 @@ import java.util.stream.Collectors;
 /**
  * Class to configure an XTDB node.
  *
- * See https://xtdb.com/reference/configuration.html for examples.
+ * See https://docs.xtdb.com/administration/configuring/ for examples.
  */
 @SuppressWarnings("unused")
 public final class NodeConfiguration {

--- a/modules/jdbc/src/xtdb/jdbc.clj
+++ b/modules/jdbc/src/xtdb/jdbc.clj
@@ -20,7 +20,7 @@
            (java.io Closeable)
            (java.sql Timestamp ResultSet)
            (java.time Duration)
-           (java.util Date)))
+           (java.util Date Map)))
 
 (defprotocol Dialect
   (setup-schema! [dialect pool])
@@ -44,14 +44,26 @@
   (close [_]
     (xio/try-close pool)))
 
+(s/def ::opts-map
+  (s/or :clj-map (s/map-of ::sys/keyword any?)
+        :java-map (s/and #(instance? Map %) #(every? string? (keys %)))))
+
+(defn- conformed-opts-map->clj-map [conformed-opts-map]
+  (let [[tag value] conformed-opts-map]
+    (case tag
+      :clj-map value
+      :java-map (zipmap (map keyword (keys value)) (vals value)))))
+
 (defn ->connection-pool {::sys/deps {:dialect nil}
                          ::sys/args {:pool-opts {:doc "Extra camelCase options to be set on HikariConfig"
-                                                 :spec (s/map-of ::sys/keyword any?)}
+                                                 :spec ::opts-map}
                                      :db-spec {:doc "db-spec to be passed to next.jdbc"
-                                               :spec (s/map-of ::sys/keyword any?)
+                                               :spec ::opts-map
                                                :required? true}}}
   [{:keys [pool-opts dialect db-spec]}]
-  (let [jdbc-url (-> (jdbcc/jdbc-url (merge {:dbtype (name (db-type dialect))} db-spec))
+  (let [pool-opts (some-> pool-opts conformed-opts-map->clj-map)
+        db-spec (some-> db-spec conformed-opts-map->clj-map)
+        jdbc-url (-> (jdbcc/jdbc-url (merge {:dbtype (name (db-type dialect))} db-spec))
                      ;; mssql doesn't like trailing '?'
                      (str/replace #"\?$" ""))
         pool-opts (merge pool-opts {:jdbcUrl jdbc-url})

--- a/test/test/xtdb/jdbc/java_configuration_test.clj
+++ b/test/test/xtdb/jdbc/java_configuration_test.clj
@@ -1,0 +1,62 @@
+(ns xtdb.jdbc.java-configuration-test
+  (:require [clojure.java.io :as io]
+            [clojure.test :as t])
+  (:import (clojure.lang IDeref)
+           (java.util HashMap Map)
+           (xtdb.api IXtdb ModuleConfiguration NodeConfiguration NodeConfiguration)))
+
+(set! *warn-on-reflection* false)
+
+(t/deftest java-map-permitted-as-db-spec-test
+  (let [dbname (name (gensym "test-db"))]
+    (-> (NodeConfiguration/builder)
+        (.with "xtdb.jdbc/connection-pool"
+               (-> (ModuleConfiguration/builder)
+                   (.with "dialect"
+                          (-> (ModuleConfiguration/builder)
+                              (.with "xtdb/module" "xtdb.jdbc.h2/->dialect")
+                              (.build)))
+                   (.set "db-spec" (doto (HashMap.) (.put "dbname" dbname)))
+                   (.build)))
+        (.with "xtdb/tx-log"
+               (-> (ModuleConfiguration/builder)
+                   (.with "xtdb/module" "xtdb.jdbc/->tx-log")
+                   (.with "connection-pool" "xtdb.jdbc/connection-pool")
+                   (.build)))
+        (.with "xtdb/document-store"
+               (-> (ModuleConfiguration/builder)
+                   (.with "xtdb/module" "xtdb.jdbc/->document-store")
+                   (.with "connection-pool" "xtdb.jdbc/connection-pool")
+                   (.build)))
+        (.build)
+        IXtdb/startNode
+        .close)
+    (t/is (.exists (io/file (format "%s.mv.db" dbname))))
+    (io/delete-file (format "%s.mv.db" dbname) true)))
+
+(t/deftest java-map-permitted-as-pool-opts-test
+  (let [dbname (name (gensym "test-db"))]
+    (with-open [node (-> (NodeConfiguration/builder)
+                         (.with "xtdb.jdbc/connection-pool"
+                                (-> (ModuleConfiguration/builder)
+                                    (.with "dialect"
+                                           (-> (ModuleConfiguration/builder)
+                                               (.with "xtdb/module" "xtdb.jdbc.h2/->dialect")
+                                               (.build)))
+                                    (.set "db-spec" (doto (HashMap.) (.put "dbname" dbname)))
+                                    (.set "pool-opts" (doto (HashMap.) (.put "maximumPoolSize" 42)))
+                                    (.build)))
+                         (.with "xtdb/tx-log"
+                                (-> (ModuleConfiguration/builder)
+                                    (.with "xtdb/module" "xtdb.jdbc/->tx-log")
+                                    (.with "connection-pool" "xtdb.jdbc/connection-pool")
+                                    (.build)))
+                         (.with "xtdb/document-store"
+                                (-> (ModuleConfiguration/builder)
+                                    (.with "xtdb/module" "xtdb.jdbc/->document-store")
+                                    (.with "connection-pool" "xtdb.jdbc/connection-pool")
+                                    (.build)))
+                         (.build)
+                         IXtdb/startNode)]
+      (t/is (= 42 (-> node :node :!system deref :xtdb.jdbc/connection-pool :pool .getMaximumPoolSize)))
+      (io/delete-file (format "%s.mv.db" dbname) true))))

